### PR TITLE
Plugins: Enforce signing for all plugins

### PR DESCRIPTION
**Pull request by wbrowne**

### CODE REVIEW

File: pkg/plugins/manager/manager_test.go [hunk 1]
```diff
@@ -61,17 +61,71 @@ func TestPluginManager_Init(t *testing.T) {
...
+	})
+
+	t.Run("With external panel plugin lacking signature (production)", func(t *testing.T) {
+		pm := createManager(t, func(pm *PluginManager) {
+			pm.Cfg.PluginsPath = "testdata/unsigned-panel"
+			pm.Cfg.Env = setting.Prod
+		})
+		err := pm.Init()
+		require.NoError(t, err)
+		pluginID := "test-panel"
```
Comment (aknuds1) [inline pkg/plugins/manager/manager_test.go:102]:
```suggestion
		const pluginID = "test-panel"
```

File: pkg/plugins/manager/manager_test.go [hunk 2]
```diff
@@ -61,17 +61,71 @@ func TestPluginManager_Init(t *testing.T) {
...
+	})
+
+	t.Run("With external back-end plugin lacking signature (development)", func(t *testing.T) {
+		pm := createManager(t, func(pm *PluginManager) {
+			pm.Cfg.PluginsPath = "testdata/unsigned-datasource"
+			pm.Cfg.Env = setting.Dev
+		})
+		err := pm.Init()
+		require.NoError(t, err)
+		pluginID := "test"
```
Comment (aknuds1) [inline pkg/plugins/manager/manager_test.go:85]:
```suggestion
		const pluginID = "test"
```

File: pkg/plugins/manager/manager_test.go [hunk 3]
```diff
@@ -61,17 +61,71 @@ func TestPluginManager_Init(t *testing.T) {
...
-	t.Run("With external back-end plugin lacking signature", func(t *testing.T) {
+	t.Run("With external back-end plugin lacking signature (production)", func(t *testing.T) {
 		pm := createManager(t, func(pm *PluginManager) {
-			pm.Cfg.PluginsPath = "testdata/unsigned"
+			pm.Cfg.PluginsPath = "testdata/unsigned-datasource"
+			pm.Cfg.Env = setting.Prod
 		})
 		err := pm.Init()
 		require.NoError(t, err)
+		pluginID := "test"
```
Comment (aknuds1) [inline pkg/plugins/manager/manager_test.go:71]:
```suggestion
		const pluginID = "test"
```

File: pkg/plugins/manager/manager_test.go [hunk 4]
```diff
@@ -87,7 +141,10 @@ func TestPluginManager_Init(t *testing.T) {
 		err := pm.Init()
 		require.NoError(t, err)
 
-		assert.Equal(t, []error{fmt.Errorf(`plugin "test" has an invalid signature`)}, pm.scanningErrors)
+		pluginID := "test"
```
Comment (aknuds1) [inline pkg/plugins/manager/manager_test.go:144]:
```suggestion
		const pluginID = "test"
```

**Review by aknuds1 — APPROVED** [review]
LGTM, but see nits.

**Review by marefr — APPROVED** [review]
LGTM

Should this be marked as a breaking change and/or highlighted in release notes and/or upgrade notes similar like [this](https://grafana.com/docs/grafana/latest/installation/upgrading/#backend-plugins)?

**Comment by wbrowne** [comment]
@marefr Just updated the docs - WDYT?

**Comment by marefr** [comment]
Looks good :+1:

**Review by achatterjee-grafana — APPROVED** [review]

**Pull request merged**